src/components/App.js

```diff
@@ -70,51 +69,34 @@ class App extends React.Component {
   };
 
   createNewBook = () => {
-    this.setState({ bookFormContent: {} });
-    this.setState({ editing: false });
+    this.setState({ selectedBook: {} });
```

Comment:
This is the performance fix isn't it. Makes sense if setState was changing the entire book list each time. Good catch!